firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -455,6 +459,122 @@ private Cursor startQuery() {
     }
   }
 
+  /**
+   * Encapsulates a query whose parameter list is so long that it might exceed SQLite limit.
+   *
+   * <p>SQLite limits maximum number of host parameters to 999 (see
+   * https://www.sqlite.org/limits.html). This class wraps most of the messy details of splitting a
+   * large query into several smaller ones.
+   */
+  static class LongQuery {
```

Comment:
Suggestions about the class name are very welcome.

Reply:
I'm fine with the `LongQuery` name but would suggest de-nesting this and just making it a top-level class.

There are a few reasons for this suggestion:

  * Consumers manipulate this type directly (`new SQLitePersistence.LongQuery(...)`)
  * You're passing the `SQLitePersistence` instance explicitly; if all classes that used a `SQLitePersistence` were nested in here this class would be absurd.
  * This class was already ~500 lines

$0.02